mkcodes.py

```diff
@@ -79,43 +113,55 @@ def get_files(inputs):
         elif path.suffix in markdown_extensions:
             yield path, path.parent
 
-def add_inits_to_dir(path):
+
+def add_inits_along_path(from_path, to_path):
     """Recursively add __init__.py files to a directory
-    This compensates for https://bugs.python.org/issue23882 and https://bugs.python.org/issue35617
+    This compensates for https://bugs.python.org/issue23882
+    and https://bugs.python.org/issue35617
     """
-    for child in path.rglob('*'):
-        if child.is_dir():
-            (child / '__init__.py').touch()
-
+    to_path = to_path.expanduser().resolve()
+    from_path = from_path.expanduser().resolve()
+    (to_path / '__init__.py').touch()
+    if to_path == from_path:
+        return
+    if to_path.relative_to(from_path):
```

Comment:
What is the purpose of this `if` clause? Shouldn't `to_path.relative_to(from_path)` always be truthy? Is it just a sanity check?

Reply:
sanity check - if later we make a mistake and somehow invoke this with a non-relative path you don't want to go up and up until we get perm denied or start dropping __init__.py in the user directory etc.